src/detect-engine-address.c

```diff
@@ -755,11 +760,13 @@ static int DetectAddressParse2(const DetectEngineCtx *de_ctx,
 
     SCLogDebug("s %s negate %s", s, negate ? "true" : "false");
 
-    for (u = 0, x = 0; u < size && x < sizeof(address); u++) {
-        if (x == (sizeof(address) - 1)) {
-            SCLogError(SC_ERR_ADDRESS_ENGINE_GENERIC, "Hit the address buffer"
-                       " limit for the supplied address.  Invalidating sig.  "
-                       "Please file a bug report on this.");
+    size_t size = strlen(s);
+    for (u = 0, x = 0; u < size && x < address_length; u++) {
+        if (x == (address_length - 1)) {
+            SCLogError(SC_ERR_ADDRESS_ENGINE_GENERIC,
+                    "Hit the address buffer"
+                    " limit for the supplied address.  Invalidating sig.  "
+                    "Please file a bug report on this.");
             goto error;
         }
         address[x] = s[u];
```

Comment:
I'm confused about the logic here, and why we'd need a 'full size' `address` here. AFAICS we use it as a temporary place to put the address from `s` before passing it on. But does it need to be so big for that? Or could we have a smaller (local) temp var that is just reset after loop interations?

Reply:
I suppose its because this parses anything between `[...]` which can have any length.